.github/workflows/build_linux_cmake.yaml

```
@@ -121,183 +128,211 @@ jobs:
 
-      - name: Upload Release assets
-        if: inputs.deploy-release
-        uses: shogo82148/actions-upload-release-asset@v1
-        with:
-          upload_url: ${{ github.event.release.upload_url }}
-          asset_path: "${{ github.workspace }}/${{ steps.lib-info.outputs.name }}/build/pkgs/librf24*"
-
-      - name: clean build environment
+      - name: Clean build environment
         if: inputs.examples-path != ''
-        working-directory: ${{ github.workspace }}/${{ steps.lib-info.outputs.name }}/build
+        working-directory: ${{ github.workspace }}/${{ github.event.repository.name }}/build
         run: sudo rm -r ./*
 
-      - name: provide ncurses
-        if: inputs.examples-path != ''
+      - name: Provide ncurses
+        # only done for native compilation because cross-compiling ncurses from
+        # source is prone to linker errors and takes a long time.
+        if: inputs.examples-path != '' && inputs.compiler == 'default'
         # This script assumes that any examples using ncurses reside in a 'ncurses' subdirectory
         # as is the case w/ RF24Mesh, RF24Gateway, and any new ncurses examples going forward.
-        run: |
-          if [[ -d ${{ github.workspace }}/${{ steps.lib-info.outputs.name }}/${{ inputs.examples-path }}/ncurses ]]; then
-            if [[ ${{ inputs.compiler }} == 'default' ]]; then
-              sudo apt-get install libncurses5-dev
-            else
-              echo "skipping the ncurses examples for cross-compiled builds"
-
-              # avoid cross-compiling linker problems in CI; this works locally but takes a while to build
-              # curl -L https://invisible-island.net/datafiles/release/ncurses.tar.gz > ncurses.tar.gz
-              # gunzip ncurses.tar.gz
-              # tar xf ncurses.tar
-              # cd ncurses-*
-              # ./configure --prefix=/usr/${{ inputs.usr-dir }} --build=${{ inputs.usr-dir }}
-              # make
-              # sudo make install
-            fi
+        env:
+          EXAMPLES_PATH: ${{ github.workspace }}/${{ github.event.repository.name }}/${{ inputs.examples-path }}/ncurses
+        run: |-
+          if [[ -d "${EXAMPLES_PATH}" ]]; then
+            sudo apt-get install libncurses5-dev
           else
             echo "ncurses is not needed for these examples"
           fi
 
-      - name: configure examples
+      - name: Configure examples
         if: inputs.examples-path != ''
-        working-directory: ${{ github.workspace }}/${{ steps.lib-info.outputs.name }}/build
-        run: |
-          cmake ../${{ inputs.examples-path }} \
-          -D CMAKE_TOOLCHAIN_FILE=../cmake/toolchains/${{ inputs.compiler }}.cmake
-
-      - name: build examples
+        working-directory: ${{ github.workspace }}/${{ github.event.repository.name }}/build
+        env:
+          # this env var is supported as of CMake v3.21+
+          CMAKE_TOOLCHAIN_FILE: cmake/toolchains/${{ inputs.compiler }}.cmake
```

```suggestion
          CMAKE_TOOLCHAIN_FILE: ../cmake/toolchains/${{ inputs.compiler }}.cmake
```
Path is relative to build/ directory, and cmake/ is an adjacent directory.